reposerver/cache/cache.go

```diff
@@ -139,14 +150,35 @@ func (c *Cache) GetGitReferences(repo string, references *[]*plumbing.Reference)
 }
 
 func manifestCacheKey(revision string, appSrc *appv1.ApplicationSource, namespace string, trackingMethod string, appLabelKey string, appName string, info ClusterRuntimeInfo) string {
+	trackingKey := trackingKey(appLabelKey, trackingMethod)
+	return fmt.Sprintf("mfst|%s|%s|%s|%s|%d", trackingKey, appName, revision, namespace, appSourceKey(appSrc)+clusterRuntimeInfoKey(info))
+}
+
+func trackingKey(appLabelKey string, trackingMethod string) string {
 	trackingKey := appLabelKey
 	if text.FirstNonEmpty(trackingMethod, string(argo.TrackingMethodLabel)) != string(argo.TrackingMethodLabel) {
 		trackingKey = trackingMethod + ":" + trackingKey
 	}
-	return fmt.Sprintf("mfst|%s|%s|%s|%s|%d", trackingKey, appName, revision, namespace, appSourceKey(appSrc)+clusterRuntimeInfoKey(info))
+	return trackingKey
 }
 
-func (c *Cache) GetManifests(revision string, appSrc *appv1.ApplicationSource, clusterInfo ClusterRuntimeInfo, namespace string, trackingMethod string, appLabelKey string, appName string, res *CachedManifestResponse) error {
+func logEntryWithManifestCacheKeyFields(revision string, appSrc *appv1.ApplicationSource, clusterInfo ClusterRuntimeInfo, namespace string, trackingMethod string, appLabelKey string, appName string, reason string) *log.Entry {
+	return log.WithFields(log.Fields{
+		"revision":    revision,
+		"appSrc":      appSourceKeyJSON(appSrc),
+		"namespace":   namespace,
+		"trackingKey": trackingKey(appLabelKey, trackingMethod),
+		"appName":     appName,
+		"clusterInfo": clusterRuntimeInfoKeyUnhashed(clusterInfo),
+		"reason":      reason,
+	})
+}
+
+func (c *Cache) GetManifests(revision string, appSrc *appv1.ApplicationSource, clusterInfo ClusterRuntimeInfo, namespace string, trackingMethod string, appLabelKey string, appName string, res *CachedManifestResponse, getReason string) error {
```

Comment:
This function is already having lots of parameters. I don't think it is a good idea to introduce an additional one just to be used in the log. It is better to log with the specific reason before calling this function.